### PR TITLE
IBX-2889: Cross content type recommendations

### PR DESCRIPTION
**Pull request by juskora**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-2889
| Versions      | 4.2, master

Steps for Dynamic targeting block will be updates as soon as the option with output type is added.

### CODE REVIEW

File: docs/personalization/cross_content_type_reco.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Cross content type recommendations
+
+Cross content type option is used to get multiple recommendation items for each input types.
```
Comment (ciastektk) [inline docs/personalization/cross_content_type_reco.md:3]:
```suggestion
Cross content type option is used to get multiple recommendation items for each output types. 
```

**Review by ciastektk — APPROVED** [review]

File: docs/personalization/cross_content_type_reco.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+To get multiple output types in the recommendation request, perform the following:
+
+1. Go to **Scenarios**.
+1. Click the **Edit** icon next to the scenario you want to set cross content type recommendations.
+1. From the **Output type** multiselect field, select the types you want to get recommendations for in the request.
+    It contains a list of all content types exported for the specified customer ID.
+1. Next, go to Preview Scenario.
+1. From the **Output type** drop-down, select **All**. 
+
+The **All** option automatically appears after content types are selected in the scenario settings.
```
Comment (dabrt) [inline docs/personalization/cross_content_type_reco.md:17]: It seems that if one has selected several output types in the edit phase, the "All" option is selected by default when you go to. preview, so there is no need to select it. You could say something like...

``` suggestion
1. From the **Output type** drop-down, select the outputs for which you want to generate the preview. 
```

And then, the sentence about the "All" option makes more sense:

"After you select content types in the scenario settings, the **All** option automatically appears here and, by default, it is selected."

However, on a second thought, it is describing the inner workings of the system, so why not simply:

"By default, the **All** option is selected."

File: docs/personalization/cross_content_type_reco.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+!!! note
+
+    The **All** option can be translated according to your needs.
+
+![Cross content type](img/perso_cross_content_type.png)
+
+## Use cross content type in Page Builder blocks
+
+When scenarios are configured to display cross content type recommendations, you can use them in the two Page Builder blocks: Dynamic targeting and Personalized.
```
Comment (dabrt) [inline docs/personalization/cross_content_type_reco.md:27]:
```suggestion
When scenarios are configured to display cross content type recommendations, you can use them in the following Page Builder blocks: Dynamic targeting and Personalized.
```

File: docs/personalization/cross_content_type_reco.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+To get all output types in the [Personalized block](../site_organization/working_with_page.md#personalized-block), in the Page Builder mode perform the following:
```
Comment (dabrt) [inline docs/personalization/cross_content_type_reco.md:29]:
```suggestion
To get all output types in the [Personalized block](../site_organization/working_with_page.md#personalized-block), in Page Builder, perform the following actions:
```

File: docs/personalization/cross_content_type_reco.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+1. In the block settings, set the scenario with configured crosscontent types output.
+1. Next, from the drop-down **Select a Content Type to be displayed**, select **All**.
+1. Increase the display limit to make sure all recommendations are shown.
+1. Click **Submit**.
```
Comment (dabrt) [inline docs/personalization/cross_content_type_reco.md:34]:
```suggestion
1. Click the **Edit** icon to display the settings of the **Personalized** block.
1. Select a scenario with cross content types output configured.
1. From the **Select a Content Type to be displayed** drop-down, select **All**.
1. Increase the display limit to ensure that all recommendations are shown.
1. Click **Submit**.
```

File: docs/personalization/cross_content_type_reco.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+
+To get all output types in the [Dynamic targeting block](../site_organization/working_with_page.md#dynamic-targeting-block):
+
+1. In the block settings, set the scenario with configured crosscontent types output.
+1. From the **Output type** drop-down, select **All**.
+
+1. Next, set the rules according to your needs.
+1. Click **Submit**.
```
Comment (dabrt) [inline docs/personalization/cross_content_type_reco.md:43]:
```suggestion
1. Click the **Edit** icon to display the settings of the **Dynamic targeting** block.
1. Select a scenario with cross content types output configured.
1. From the **Output type** drop-down, select **All**.
1. Set the matching rules according to your needs.
1. Click **Submit**.
```

File: docs/personalization/scenarios.md [hunk 7]
```diff
@@ -25,8 +25,7 @@ If this is the case, use the selector field to switch between views for each of
 ## Content type configuration
 
 Every scenario supports a single input type and multiple output types. 
-Every recommendation request delivers only content of one output content type 
-(even if multiple types are selected in the interface below). 
+Every recommendation request can deliver content of one or all output content types.
```
Comment (dabrt) [inline docs/personalization/scenarios.md:28]: How about selecting two out of three available content types? If this is doable, I'd go with: 

```suggestion
Every recommendation request can deliver content of one or more output content types.
```

File: docs/site_organization/working_with_page.md [hunk 8]
```diff
@@ -147,6 +147,7 @@ On the **Basic** tab, perform the following actions:
 - In the **Select default scenario** section, select the default scenario for recommended items that should be rendered if the current user
 is not assigned to any Segment.
 - In the **Setup segment and scenario matching priority rules** drop-down, select a Segment group, a Segment identifier and Scenario that you want to display recommendations from.
+- **Display limit** – Set the number of products to be displayed, for example, 4.
```
Comment (dabrt) [inline docs/site_organization/working_with_page.md:150]:
```suggestion
- In the **Display limit** field, set the number of recommended items to be displayed, for example, 4.
```

**Review by dabrt — CHANGES_REQUESTED** [review]

**Review by dabrt — APPROVED** [review]

**Pull request merged**